src/ResultWriter/WaveFieldWriter.h

```diff
@@ -248,6 +258,33 @@ class WaveFieldWriter : private async::Module<WaveFieldWriterExecutor, WaveField
 			}
 		}
 
+		if (m_integrals) {
+			unsigned int offset = 0;
+			unsigned int flagOffset = WaveFieldWriterExecutor::NUM_LOWVARIABLES;
+		if (m_pstrain) {
+				offset = WaveFieldWriterExecutor::NUM_LOWVARIABLES;
+				flagOffset = 0;
+			}
+
+			nextId = offset;
+			for (unsigned int i = offset; i < offset+WaveFieldWriterExecutor::NUM_INTEGRATED_VARIABLES; i++) {
```

Comment:
Iterate form 0 to `WaveFieldWriterExecutor::NUM_INTEGRATED_VARIABLES`. You can remove `flagOffset` completely and always use `WaveFieldWriterExecutor::NUM_LOWVARIABLES`.